proxy/ProxyClientTransaction.h

```diff
@@ -174,10 +174,10 @@ class ProxyClientTransaction : public VConnection
     return true;
   }
 
+  virtual void destroy();
   virtual void
-  destroy()
+  transaction_done()
```

Comment:
Consider a more explicit tracking of the transaction state, even a simple IDLE, ACTIVE, DONE would probably make things a lot more explicit.